### ISSUE TITLE
ncurry Bug

### ISSUE THREAD

**Issue by moonglum** [beastaugh/udon#2]
I wanted to test ncurry, I ran into the following error when running the according example (in Safari):

```
function add3(a, b, c) {
    return a + b + c;
}

var curry3 = Udon.ncurry(3),
    add3c  = curry3(add3);

/* curry is now:
function (f) {
        var a1 = Udon._slice.call(arguments, 1),
        accumulator = function() {
            var a2 = a1.concat(Udon._slice.call(arguments, 0));
            if (a2.length < n) {
                a2.unshift(f);
                a1 = a2;
                return accumulator;
            } else {
                return f.apply(null, a2);
            }
        };
        return accumulator;
    }
*/

/* add3c is now:
function () {
            var a2 = a1.concat(Udon._slice.call(arguments, 0));
            if (a2.length < n) {
                a2.unshift(f);
                a1 = a2;
                return accumulator;
            } else {
                return f.apply(null, a2);
            }
        }
*/

add3c(1)(2)(3); 
/* ->
TypeError: Result of expression 'add3c(1)(2)' [function add3(a, b, c) {
        return a + b + c;
    }12] is not a function.
*/
add3c(1)(2);
/* ->
TypeError: Result of expression 'add3c(1)' [function add3(a, b, c) {
        return a + b + c;
    }11] is not a function.*/
add3c(1)
/* 
"function add3(a, b, c) {
        return a + b + c;
    }11"
*/
```

**Comment by beastaugh** [comment]
Thanks! Writing a test suite is top of my to-do list for this project, I'll make sure this gets in there.

**Comment by moonglum** [comment]
Which library did you think about? I made good experience with [QUnit](http://docs.jquery.com/Qunit). Even though it is written by the jQuery guys, it is absolutely useful for testing plain JavaScript as well.

**Comment by beastaugh** [comment]
Not sure yet—I was wondering about using [qc.js](http://willowbend.cx/2009/12/05/qc-js-quickcheck-javascript/).

**Comment by beastaugh** [comment]
This should be fixed in [this commit](1cae951eaed726d0bcc528145ed36c0075762ab3).
